client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { localize } from 'i18n-calypso';
+import Button from 'components/button';
+import Card from 'components/card';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormToggle from 'components/forms/form-toggle';
+import FormRadio from 'components/forms/form-radio';
+import FormTextInput from 'components/forms/form-text-input';
+import Notice from 'components/notice';
+import QueryMailChimpSyncStatus from 'woocommerce/state/sites/settings/email/querySyncStatus';
+import {
+	syncStatus,
+	mailchimpSettings,
+	isRequestingSettings } from 'woocommerce/state/sites/settings/email/selectors';
+import { submitMailchimpNewsletterSettings, requestResync } from 'woocommerce/state/sites/settings/email/actions.js';
+import { isSubmittingNewsletterSetting, newsletterSettingsSubmitError } from 'woocommerce/state/sites/settings/email/selectors';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+const SyncTab = localize( ( { siteId, translate, syncState, resync } ) => {
+	const { account_name, store_syncing, product_count, mailchimp_total_products,
+		mailchimp_total_orders, order_count } = syncState;
+	const hasProductInfo = ( product_count !== undefined ) && ( mailchimp_total_products !== undefined );
+	const products = hasProductInfo ? ( product_count + '/' + mailchimp_total_products ) : 'X/X';
+	const hasOrdersInfo = ( order_count !== undefined ) && ( mailchimp_total_orders !== undefined );
+	const orders = hasOrdersInfo ? ( order_count + '/' + mailchimp_total_orders ) : 'X/X';
+
+	const synced = () => (
+		<Notice
+			status="is-success"
+			isCompact={ true }
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list synced' ) }>
+		</Notice>
+	);
+
+	const syncing = () => (
+		<Notice
+			status="is-warning"
+			isCompact={ true }
+			showDismiss={ false }
+			text={ syncState.mailchimp_list_name + ' ' + translate( 'list is being synced' ) }>
+		</Notice>
+	);
+
+	const onResyncCLick = () => {
+		resync( siteId );
+	};
+
+	return (
+		<div>
+			<div>
+				<span className="mailchimp__account-info">{ translate( 'MailChimp account:' ) }</span>
```

Comment:
I think we need to have a loading indicator here so we can show something before the mailing list info is ready. I get a flash of 'undefined' before the actual list name displays.

Reply:
@kellychoffman ^ how would you like me to tackle this?

Reply:
If you detect that the info is still loading, you can instead show a placeholder div.
That div can include some CSS that will show a pulsing empty div like we do in other places of the app. Example: https://github.com/Automattic/wp-calypso/blob/782eb8610aca534ec92aaf035b7cbfa34cd9d4c3/client/extensions/woocommerce/components/store-address/style.scss#L1-L5